code/game/machinery/poolcontroller.dm

```diff
@@ -39,63 +42,73 @@
-				continue
-			if(M.stat == DEAD)
-				continue
-			//End sanity checks, go on
-			switch(temperature) //Apply different effects based on what the temperature is set to.
-				if("scalding") //Burn the mob.
-					M.bodytemperature = min(500, M.bodytemperature + 35) //heat mob at 35k(elvin) per cycle
-					M << "<span class='danger'>The water is searing hot!</span>"
-
-				if("warm") //Gently warm the mob.
-					M.bodytemperature = min(330, M.bodytemperature + 10) //Heats up mobs to just over normal, not enough to burn
-					if(prob(50)) //inform the mob of warm water half the time
-						M << "<span class='warning'>The water is quite warm.</span>" //Inform the mob it's warm water.
-
-				if("cool") //Gently cool the mob.
-					M.bodytemperature = max(290, M.bodytemperature - 10) //Cools mobs to just below normal, not enough to burn
-					if(prob(50)) //inform the mob of cold water half the time
-						M << "<span class='warning'>The water is chilly.</span>" //Inform the mob it's chilly water.
-
-				if("frigid") //Freeze the mob.
-					M.bodytemperature = max(80, M.bodytemperature - 35) //cool mob at -35k per cycle
-					M << "<span class='danger'>The water is freezing!</span>"
-
+	for(var/turf/T in linkedturfs) //Check for pool-turfs linked to the controller.
+		for(var/mob/M in T) //Check for mobs in the linked pool-turfs.
+			if(temperature != "normal")		//don't bother with this if we aren't going to be changing anything anyways
+				handleTemp(M)
 
 			if(ishuman(M)) //Make sure they are human before typecasting.
 				var/mob/living/carbon/human/drownee = M //Typecast them as human.
-				if(drownee && drownee.lying) //Mob lying down
-					if(drownee.internal)
-						continue //Has internals, no drowning
-					if((drownee.species.flags & NO_BREATHE) || (NO_BREATHE in drownee.mutations))
-						continue //doesn't breathe, no drowning
-					if(drownee.get_species() == "Skrell" || drownee.get_species() == "Neara")
-						continue //fish things don't drown
-
-					if(drownee.stat) //Mob is in critical.
-						drownee.adjustOxyLoss(9) //Kill em quickly.
-						add_logs(drownee, src, "drowned")
-						drownee.visible_message("<span class='danger'>\The [drownee] appears to be drowning!</span>","<span class='userdanger'>You're quickly drowning!</span>") //inform them that they are fucked.
-					else
-						drownee.adjustOxyLoss(5) //5 oxyloss per cycle.
-						add_logs(drownee, src, "drowned")
-						if(prob(35)) //35% chance to tell them what is going on. They should probably figure it out before then.
-							drownee.visible_message("<span class='danger'>\The [drownee] flails, almost like they are drowning!</span>","<span class='userdanger'>You're lacking air!</span>") //*gasp* *gasp* *gasp* *gasp* *gasp*
-
-		for(var/obj/effect/decal/cleanable/decal in W)
+				handleDrowning(drownee)		//Only human types will drown, to keep things simple for non-human mobs that live in the water
+
+		for(var/obj/effect/decal/cleanable/decal in T)
```

Comment:
add comments explaining what each thing does please


no it's not really necessary but I spent about an hour making the thing as easy to understand for someone new as possible so do it